src/generator.cs

```diff
@@ -1485,7 +1485,7 @@ public partial class Generator : IMemberGatherer {
 	Dictionary<Type,TrampolineInfo> trampolines = new Dictionary<Type,TrampolineInfo> ();
 	Dictionary<Type,int> trampolines_generic_versions = new Dictionary<Type,int> ();
 	Dictionary<Type,Type> notification_event_arg_types = new Dictionary<Type,Type> ();
-	List <string> libraries = new List <string> ();
+	List<Tuple<string, string>> libraries = new List<Tuple<string, string>> (); // <LibraryName, libraryPath>
 
```

Comment:
I've got a small question about this. You are using a Tuple that contains the lib name and the lib path. Later in the code we have to use things such as:

```csharp

// searching for the lib
if (!libraries.Any (l => l.Item1 == library_name)) {
    libraries.Add (new Tuple<string, string> (library_name, library_path));
}

// looping for the libs 
foreach (var library_info in libraries.OrderBy (v => v.Item1))
```

I'm wondering how many times (if ever) we have a LibraryName that is not unique. If that is guaranteed, that is, no library will have the same name as other, using a dictionary (the StringDictionary specialised collection) over a List <Tuple <string, string>> makes sense. Searching will be faster, we will have less line operations (and we all know that is good ;) ) and makes things such as accessing the data cleaner removing things like:

```csharp
var library_name = library_info.Item1; // Getting the info from the tuple for readability
 var library_path = library_info.Item2;
```